code/client/src/components/AddressInput.jsx

```diff
@@ -163,39 +163,50 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
 
       const unlabelledWalletAddress = wallets.filter(w => existingKnownAddresses.find(a => !a.label && !a.domain?.name && a.address === w.address && (!w.temp || allowTemp)))
 
-      // Init the known address entries for existing wallets.
-      walletsNotInKnownAddresses.forEach((wallet) => {
-        dispatch(globalActions.setKnownAddress({
-          label: wallet.name,
-          address: wallet.address,
-          network: wallet.network,
-          creationTime: wallet.effectiveTime,
-          numUsed: 0
-        }))
-      })
-
-      unlabelledWalletAddress.forEach((w) => {
-        dispatch(globalActions.setKnownAddress({
-          ...knownAddresses[w],
-          label: w.name,
-        }))
-      })
-
-      await Promise.all(knownAddressesWithoutDomain.map(async (knownAddress) => {
+      const domainWalletAddresses = await Promise.all(knownAddressesWithoutDomain.map(async (knownAddress) => {
```

Comment:
There could be a long delay in resolving this. We shouldn't wait for this to finish before updating the rest

Reply:
Done